app/javascript/controllers/color_avatar_controller.js

```diff
@@ -3,13 +3,35 @@ import { Controller } from "@hotwired/stimulus";
 // Connects to data-controller="color-avatar"
 // Used by the transaction merchant form to show a preview of what the avatar will look like
 export default class extends Controller {
-  static targets = ["name", "avatar", "selection"];
+  static targets = ["name", "avatar", "selection","colorInput"];
 
   connect() {
-    this.nameTarget.addEventListener("input", this.handleNameChange);
+    if(!this.hasColorInputTarget){
+      this.nameTarget.addEventListener("input", this.handleNameChange);
+    }
+
+    if(this.hasColorInputTarget){
+      this.observer = new MutationObserver((mutations) => {
+        mutations.forEach((mutation) => {
+          if (mutation.type === "attributes" && mutation.attributeName === "data-color-picker-color-value") {
+            const color = mutation.target.value;
+            this.updateAvatarColors(color);
+          }
+        });
+      });
+  
+      this.observer.observe(this.colorInputTarget, {
+        attributes: true,
+        attributeFilter: ["data-color-picker-color-value"]
+      });
+    }
```

Comment:
I think we should avoid observers like this and update this with one of the following approaches:

1. Combine everything into a single controller (probably the easiest)
2. Use Stimulus outlets to update:

```js
static outlets = ["color-avatar"];

// ...

initPicker() {
 // ...
 this.picker.on("change", (color, source, instance) => {
   const hexColor = color.toHEXA().toString();
   this.colorAvatarOutlet.updateAvatarColors(hexColor);
 }
}
```